src/operations.cpp

```diff
@@ -1399,6 +1419,67 @@ void last_write_time(const path& p, const std::time_t new_time,
 # endif
 }
 
+BOOST_FILESYSTEM_DECL
+std::time_t creation_time(const path& p, system::error_code* ec)
+{
+# ifdef BOOST_POSIX_API
+# ifdef __linux__
+  struct statx path_stat;
+  if (error(::statx(AT_FDCWD, p.c_str(), AT_SYMLINK_NOFOLLOW | AT_NO_AUTOMOUNT |
+	AT_STATX_SYNC_AS_STAT, STATX_BTIME, &path_stat)!= 0 ? BOOST_ERRNO : 0,
+    p, ec, "boost::filesystem::creation_time"))
+      return std::time_t(-1);
+  return std::time_t(path_stat.stx_btime.tv_sec);
+# elif defined(__APPLE__)
+  // The stat64() system calls first appeared in Mac OS X 10.5 (Leopard)
+#   if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
+#     ifdef _DARWIN_USE_64_BIT_INODE
+  struct stat path_stat;
+  if (error(::stat(p.c_str(), &path_stat) != 0 ? BOOST_ERRNO : 0, p, ec,
+	"boost::filesystem::creation_time"))
+      return std::time_t(-1);
+  return std::time_t(path_stat.st_birthtimespec.tv_sec);
+#     else
+  struct stat64 path_stat;
+  if (error(::stat64(p.c_str(), &path_stat) != 0 ? BOOST_ERRNO : 0, p, ec,
+	"boost::filesystem::creation_time"))
+      return std::time_t(-1);
+  return std::time_t(path_stat.st_birthtimespec.tv_sec);
+#     endif
+#   endif
+# elif defined(__FreeBSD__)
+  struct stat path_stat;
+  if (error(::stat(p.c_str(), &path_stat) != 0 ? BOOST_ERRNO : 0, p, ec,
+	"boost::filesystem::creation_time"))
+      return std::time_t(-1);
+  return std::time_t(path_stat.st_birthtim.tv_sec);
```

Comment:
I think, `st_birthtime` is a more portable way to access this field instead of `st_birthtim.tv_sec`. Also, NetBSD seems to support it as well: https://netbsd.gw.com/cgi-bin/man-cgi?stat+2.i386+NetBSD-8.0

Reply:
it is only work, when _POSIX_SOURCE not defined, so I think it is dangerous to use this.

Please see https://www.freebsd.org/cgi/man.cgi?query=stat&sektion=2